File: reference/SmartPharmacy-Kotlin/app/src/main/java/com/example/ui/screens/ReportsScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Assessment
import androidx.compose.material.icons.filled.Download
import androidx.compose.material.icons.outlined.Inventory2
import androidx.compose.material.icons.outlined.PointOfSale
import androidx.compose.material.icons.outlined.ShoppingBag
import androidx.compose.material.icons.outlined.Warning
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.DashboardCard
import com.example.ui.components.SectionHeader
import com.example.ui.theme.*
import com.example.ui.viewmodel.PharmacyViewModel

@Composable
fun ReportsScreen(
    viewModel: PharmacyViewModel
) {
    val medicines by viewModel.medicines.collectAsState()
    val sales by viewModel.sales.collectAsState()
    val purchases by viewModel.purchases.collectAsState()

    val totalSales = sales.sumOf { it.grandTotal }
    val totalPurchases = purchases.sumOf { it.totalAmount }
    val totalProducts = medicines.size
    val expiredCount = viewModel.getExpiredMedicines(medicines).size

    // Monthly Chart Data (Jan - Jul 2026)
    val months = listOf("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul")
    val salesData = listOf(125000f, 142000f, 138000f, 165000f, 180000f, 172000f, 195000f)
    val purchaseData = listOf(85000f, 92000f, 105000f, 98000f, 110000f, 115000f, 120000f)

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(SurfaceWhite)
            .padding(16.dp)
            .verticalScroll(rememberScrollState()),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Top Banner
        Card(
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(containerColor = PrimaryBlue),
            modifier = Modifier.fillMaxWidth()
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(18.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(text = "Pharmacy Analytics & Reports", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color.White)
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(text = "Financial performance & inventory metrics", fontSize = 12.sp, color = Color.White.copy(alpha = 0.85f))
                }
                Box(
                    modifier = Modifier
                        .size(44.dp)
                        .clip(CircleShape)
                        .background(Color.White.copy(alpha = 0.2f)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(imageVector = Icons.Default.Assessment, contentDescription = null, tint = Color.White)
                }
            }
        }

        SectionHeader(title = "Financial Summary")

        // Metric Cards Grid
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            DashboardCard(
                title = "Total Sales",
                value = "₹${(totalSales / 1000).toInt()}k",
                icon = Icons.Outlined.PointOfSale,
                iconBgColor = SuccessGreenLight,
                iconColor = SuccessGreen,
                modifier = Modifier.weight(1f)
            )

            DashboardCard(
                title = "Total Purchases",
                value = "₹${(totalPurchases / 1000).toInt()}k",
                icon = Icons.Outlined.ShoppingBag,
                iconBgColor = SecondaryTealLight,
                iconColor = SecondaryTeal,
                modifier = Modifier.weight(1f)
            )
        }

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            DashboardCard(
                title = "Active Products",
                value = "$totalProducts",
                icon = Icons.Outlined.Inventory2,
                iconBgColor = PrimaryBlueLight,
                iconColor = PrimaryBlue,
                modifier = Modifier.weight(1f)
            )

            DashboardCard(
                title = "Expired Count",
                value = "$expiredCount",
                icon = Icons.Outlined.Warning,
                iconBgColor = AlertRedLight,
                iconColor = AlertRed,
                modifier = Modifier.weight(1f)
            )
        }

        SectionHeader(title = "Monthly Sales vs Purchases (2026)")

        // Interactive Canvas Bar Chart
        Card(
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
            elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
            modifier = Modifier
                .fillMaxWidth()
                .border(1.dp, SurfaceBorder, RoundedCornerShape(16.dp))
        ) {
            Column(modifier = Modifier.padding(18.dp)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.End,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Box(modifier = Modifier.size(10.dp).background(SuccessGreen, CircleShape))
                    Spacer(modifier = Modifier.width(4.dp))
                    Text("Sales", fontSize = 11.sp, color = TextSecondary)

                    Spacer(modifier = Modifier.width(12.dp))

                    Box(modifier = Modifier.size(10.dp).background(SecondaryTeal, CircleShape))
                    Spacer(modifier = Modifier.width(4.dp))
                    Text("Purchases", fontSize = 11.sp, color = TextSecondary)
                }

                Spacer(modifier = Modifier.height(16.dp))

                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(180.dp)
                ) {
                    Canvas(modifier = Modifier.fillMaxSize()) {
                        val canvasWidth = size.width
                        val canvasHeight = size.height
                        val maxVal = 220000f

                        val groupWidth = canvasWidth / months.size
                        val barWidth = groupWidth * 0.35f

                        for (i in months.indices) {
                            val groupX = i * groupWidth + (groupWidth - (barWidth * 2 + 8)) / 2

                            // Sales Bar
                            val salesBarHeight = (salesData[i] / maxVal) * (canvasHeight - 30)
                            drawRect(
                                color = SuccessGreen,
                                topLeft = Offset(groupX, canvasHeight - salesBarHeight - 20),
                                size = Size(barWidth, salesBarHeight)
                            )

                            // Purchases Bar
                            val purchaseBarHeight = (purchaseData[i] / maxVal) * (canvasHeight - 30)
                            drawRect(
                                color = SecondaryTeal,
                                topLeft = Offset(groupX + barWidth + 6, canvasHeight - purchaseBarHeight - 20),
                                size = Size(barWidth, purchaseBarHeight)
                            )
                        }
                    }
                }

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceAround
                ) {
                    months.forEach { month ->
                        Text(text = month, fontSize = 11.sp, color = TextSecondary, fontWeight = FontWeight.SemiBold)
                    }
                }
            }
        }
    }
}

File: lib/screens/reports_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../core/theme/app_colors.dart';
import '../providers/pharmacy_provider.dart';
import '../widgets/common/section_header.dart';
import '../widgets/dashboard/dashboard_card.dart';

class ReportsScreen extends StatelessWidget {
  const ReportsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    
    final totalSales = provider.sales.fold(0.0, (sum, s) => sum + s.grandTotal);
    final totalPurchases = provider.purchases.fold(0.0, (sum, p) => sum + p.totalAmount);
    final totalProducts = provider.medicines.length;
    final expiredCount = provider.expiredMedicines.length;

    // Monthly Chart Data (Jan - Jul 2026) matches Kotlin reference
    final months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul"];
    final salesData = [125000.0, 142000.0, 138000.0, 165000.0, 180000.0, 172000.0, 195000.0];
    final purchaseData = [85000.0, 92000.0, 105000.0, 98000.0, 110000.0, 115000.0, 120000.0];

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(16.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // Top Banner
              _buildTopBanner(),
              const SizedBox(height: 16),

              const SectionHeader(title: "Financial Summary"),
              const SizedBox(height: 8),

              // Metric Cards Grid
              Row(
                children: [
                  Expanded(
                    child: DashboardCard(
                      title: "Total Sales",
                      value: "₹${(totalSales / 1000).toInt()}k",
                      icon: Icons.point_of_sale_outlined,
                      iconBgColor: AppColors.successGreenLight,
                      iconColor: AppColors.successGreen,
                    ),
                  ),
                  const SizedBox(width: 10),
                  Expanded(
                    child: DashboardCard(
                      title: "Total Purchases",
                      value: "₹${(totalPurchases / 1000).toInt()}k",
                      icon: Icons.shopping_bag_outlined,
                      iconBgColor: AppColors.secondaryTealLight,
                      iconColor: AppColors.secondaryTeal,
                    ),
                  ),
                ],
              ),
              const SizedBox(height: 10),
              Row(
                children: [
                  Expanded(
                    child: DashboardCard(
                      title: "Active Products",
                      value: "$totalProducts",
                      icon: Icons.inventory_2_outlined,
                      iconBgColor: AppColors.primaryBlueLight,
                      iconColor: AppColors.primaryBlue,
                    ),
                  ),
                  const SizedBox(width: 10),
                  Expanded(
                    child: DashboardCard(
                      title: "Expired Count",
                      value: "$expiredCount",
                      icon: Icons.warning_amber_outlined,
                      iconBgColor: AppColors.alertRedLight,
                      iconColor: AppColors.alertRed,
                    ),
                  ),
                ],
              ),
              const SizedBox(height: 16),

              const SectionHeader(title: "Monthly Sales vs Purchases (2026)"),
              const SizedBox(height: 8),

              // Interactive Bar Chart Card
              _buildChartCard(months, salesData, purchaseData),
              const SizedBox(height: 16),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildTopBanner() {
    return Card(
      elevation: 0,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
      color: AppColors.primaryBlue,
      child: Padding(
        padding: const EdgeInsets.all(18.0),
        child: Row(
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: const [
                  Text(
                    "Pharmacy Analytics & Reports",
                    style: TextStyle(
                      fontSize: 18,
                      fontWeight: FontWeight.bold,
                      color: Colors.white,
                    ),
                  ),
                  SizedBox(height: 4),
                  Text(
                    "Financial performance & inventory metrics",
                    style: TextStyle(
                      fontSize: 12,
                      color: Colors.white70,
                    ),
                  ),
                ],
              ),
            ),
            Container(
              width: 44,
              height: 44,
              decoration: BoxDecoration(
                color: Colors.white.withValues(alpha: 0.2),
                shape: BoxShape.circle,
              ),
              child: const Icon(
                Icons.assessment,
                color: Colors.white,
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildChartCard(List<String> months, List<double> salesData, List<double> purchaseData) {
    return Card(
      elevation: 2,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: const BorderSide(color: AppColors.surfaceBorder),
      ),
      color: AppColors.surfaceWhite,
      child: Padding(
        padding: const EdgeInsets.all(18.0),
        child: Column(
          children: [
            Row(
              mainAxisAlignment: MainAxisAlignment.end,
              children: [
                _buildLegendItem("Sales", AppColors.successGreen),
                const SizedBox(width: 12),
                _buildLegendItem("Purchases", AppColors.secondaryTeal),
              ],
            ),
            const SizedBox(height: 16),
            SizedBox(
              height: 180,
              width: double.infinity,
              child: CustomPaint(
                painter: BarChartPainter(
                  salesData: salesData,
                  purchaseData: purchaseData,
                  months: months,
                ),
              ),
            ),
            const SizedBox(height: 8),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceAround,
              children: months.map((month) => Text(
                month,
                style: const TextStyle(fontSize: 11, color: AppColors.textSecondary, fontWeight: FontWeight.w600),
              )).toList(),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildLegendItem(String label, Color color) {
    return Row(
      children: [
        Container(
          width: 10,
          height: 10,
          decoration: BoxDecoration(color: color, shape: BoxShape.circle),
        ),
        const SizedBox(width: 4),
        Text(
          label,
          style: const TextStyle(fontSize: 11, color: AppColors.textSecondary),
        ),
      ],
    );
  }
}

class BarChartPainter extends CustomPainter {
  final List<double> salesData;
  final List<double> purchaseData;
  final List<String> months;

  BarChartPainter({
    required this.salesData,
    required this.purchaseData,
    required this.months,
  });

  @override
  void paint(Canvas canvas, Size size) {
    final maxVal = 220000.0;
    final groupWidth = size.width / months.length;
    final barWidth = groupWidth * 0.35;
    final spacing = 6.0;

    final paintSales = Paint()..color = AppColors.successGreen;
    final paintPurchases = Paint()..color = AppColors.secondaryTeal;

    for (int i = 0; i < months.length; i++) {
      final groupX = i * groupWidth + (groupWidth - (barWidth * 2 + spacing)) / 2;

      // Sales Bar
      final salesBarHeight = (salesData[i] / maxVal) * (size.height - 30);
      canvas.drawRect(
        Rect.fromLTWH(
          groupX,
          size.height - salesBarHeight - 20,
          barWidth,
          salesBarHeight,
        ),
        paintSales,
      );

      // Purchases Bar
      final purchaseBarHeight = (purchaseData[i] / maxVal) * (size.height - 30);
      canvas.drawRect(
        Rect.fromLTWH(
          groupX + barWidth + spacing,
          size.height - purchaseBarHeight - 20,
          barWidth,
          purchaseBarHeight,
        ),
        paintPurchases,
      );
    }
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) => false;
}
